### PR TITLE
Personal loan lw

### PR DESCRIPTION
**Pull request by ros3bud**
@cgrayson, this is includes the factoring out of the response method as we discussed. LGTY?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ module.exports = {
   inbound: {
     shortform: require('./lib/shortform'),
     unified: require('./lib/unified'),
-    multiformat: require('./lib/multiformat')
+    multiformat: require('./lib/multiformat'),
+    personal_loan: require('./lib/personalloan')
```
Comment (cgrayson) [inline index.js]: Minor, but let's have these alphabetized.

File: src/personalloan.coffee [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    filter_name: lead.PartnerProfileInformation.FilterName
+    filter_routing_id: lead.PartnerProfileInformation.FilterRoutingID
+    routing_parameters: lead.PartnerProfileInformation.RoutingParams
+    filter_class: lead.PartnerProfileInformation.FilterCategory
+
+
+request.variables = ->
+[
+  { name: 'first_name', type: 'string', description: 'First name' },
+  { name: 'last_name', type: 'string', description: 'Last name' },
```
Comment (cgrayson) [inline src/personalloan.coffee]: You could remove these commas, like you did in `shortform.coffee`. :smiley:

File: src/personalloan.coffee [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  { name: 'email', type: 'string', description: 'Email address' },
+  { name: 'phone_1', type: 'string', description: 'Phone 1 (home)' },
+  { name: 'address_1', type: 'string', description: 'Address 1' },
+  { name: 'city', type: 'string', description: 'City' },
+  { name: 'state', type: 'string', description: 'State' },
+  { name: 'postal_code', type: 'string', description: 'Postal code' },
```
Comment (cgrayson) [inline src/personalloan.coffee]: These request variables (`postal_code`, `city`, `state`, etc.) can have types. (Our talk today about only using the `string` type doesn’t apply to inbound.)

File: src/personalloan.coffee [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+getPhone1 = (areaCode, phone, masked, extension) ->
+  return null unless areaCode or phone or extension
+  if masked is "Y"
+    phone = "#{areaCode}*******"
+  else
+    phone = "#{areaCode}#{phone}"
+  
+  if extension
+      phone += " x#{extension}"
```
Comment (cgrayson) [inline src/personalloan.coffee]: I wonder whether this should also be `*`'d out, if `masked` is "Y"? (Also, there's an extra indent on this line.)

File: src/personalloan.coffee [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+
+#
+# Helpers ----------------------------------------------------------------
+#
+
+
+getPhone1 = (areaCode, phone, masked, extension) ->
+  return null unless areaCode or phone or extension
+  if masked is "Y"
```
Comment (cgrayson) [inline src/personalloan.coffee]: I wonder if it's worth being extra careful and comparing `masked.toLowerCase()` to "y" here.

File: spec/personalloan_spec.coffee [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+        assert.equal personalloan.request(req(xml())).loan.amount, '3250000'
+
+      it 'should parse term_in_months', ->
+        assert.equal personalloan.request(req(xml())).loan.term_in_months, '120'
+
+      it 'should parse is_existing_customer', ->
+        assert.equal personalloan.request(req(xml())).loan.is_existing_customer, 'N'
+
+
+  describe 'LendingTree', ->
```
Comment (cgrayson) [inline spec/personalloan_spec.coffee]: Seems like this should be "under" `Describe 'Lead'`, no?

File: spec/personalloan_spec.coffee [hunk 7]
```diff
@@ -0,0 +1,292 @@
+assert    = require('chai').assert
+moment    = require('moment')
+personalloan = require '../src/personalloan'
+fields    = require('leadconduit-fields')
```
Comment (cgrayson) [inline spec/personalloan_spec.coffee]: I don't see where `moment` or `fields` are used here.

**Comment by cgrayson** [comment]
This looks good, with the issues noted for you to consider. 

Having that common response code pulled out is great. I had a couple of thoughts about improving that even more. One, I think the code in `parse.coffee` could move into `common.coffee`. Minor, maybe, but the parsing functions are indeed "common", after all. Also, since the `response` code has moved into a common file, all the `response`-related _tests_ could (/should) also be moved into one `common_spec` file. (If you'd like to address these further refactoring efforts in another PR after finishing this new inbound integration, I think that's fine.)

File: src/personalloan.coffee [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+  return null unless value
+  return 60 if value is '1'
+  return 84 if value is '2'
+  return 120 if value is '3'
+  return 180 if value is '4'
+  return 240 if value is '5'
+  return 300 if value is '6'
+  return 360 if value is '7'
+  return 480 if value is '8'
+  return value
```
Comment (alexkwolfe) [inline src/personalloan.coffee]: I wonder if these would be more succinct as a switch statements?

```coffeescript
parseMonths = (value) ->
  return null unless value
  switch value
    when '1' then 60
    when '2' then 84
    when '3' then 120
    when '4' then 180
    # ...
    else value
```

File: index.js (hunk 1 above)
Comment (ros3bud) [inline index.js]: Ok - no problem.

File: src/personalloan.coffee (hunk 2 above)
Comment (ros3bud) [inline src/personalloan.coffee]: Ack, for sure. That was an experiment gone wrong.

File: src/personalloan.coffee (hunk 3 above)
Comment (ros3bud) [inline src/personalloan.coffee]: ah, ok.

File: src/personalloan.coffee (hunk 4 above)
Comment (ros3bud) [inline src/personalloan.coffee]: I thought about that, too. My thinking was that without the phone number, the extension is pretty worthless. Also, it does not come in as masked like phone, so this way we are reflecting the data we receive. Do you think that is enough of a reason to keep it unmasked?

File: src/personalloan.coffee (hunk 5 above)
Comment (ros3bud) [inline src/personalloan.coffee]: It definitely can't hurt.

File: spec/personalloan_spec.coffee (hunk 6 above)
Comment (ros3bud) [inline spec/personalloan_spec.coffee]: si.

File: src/personalloan.coffee (hunk 8 above)
Comment (ros3bud) [inline src/personalloan.coffee]: That's a good point.

**Comment by cgrayson** [comment]
LGTM. As far as my thoughts on improvements, what I wrote before was all I had. :smiley:

**Pull request merged**